examples/vision/image_classification_with_vision_transformer.py

```diff
@@ -0,0 +1,294 @@
+"""
+Title: Image Classification with Vision Transformer
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/01/30
+Last modified: 2021/01/30
+Description: Implementing Vision Transformer (ViT) model for image classification.
+"""
+
+"""
+## Introduction
+
+This example implements the [Vision Transformer (ViT)](https://arxiv.org/abs/2010.11929)
+model, by Alexey Dosovitskiy et al., for CIFAR-100 image classification.
+The ViT applies the transformer architecture with self-attentions to sequences of
+image patches, without using convolutional networks.
+
+The example requires TensorFlow 2.4 or higher, and
+[TensorFlow Addons](https://www.tensorflow.org/addons/overview),
+which can be installed using the following command:
```

Comment:
Suggestion:

- Not sure if the extra commas are necessary here.

```suggestion
This example implements the [Vision Transformer (ViT)](https://arxiv.org/abs/2010.11929)
model by Alexey Dosovitskiy et al. for image classification on the CIFAR-100 dataset.
The ViT applies the Transformer architecture with self-attentions on sequences of
image patches without using convolutional networks.

The example requires TensorFlow 2.4 or higher and
[TensorFlow Addons](https://www.tensorflow.org/addons/overview),
which can be installed using the following command:
```

Reply:
Just to double check - is this 100% your work? If you have used some open-sourced code, maybe it'd be a good idea to reference them and give the original authors credit (even if you used their, say, PyTorch code to implement your TensorFlow version).

Reply:
Yes indeed, it is 100% my work. The original implementation of the paper's authors is in JAX. I found other implementations online, but this one is different in many ways, as I try to simplify, use existing APIs, and follow the best practices used in other examples. In addition, I made a few changes to the original implementation.

Reply:
that's amazing 👍